fix(serverless-api): retries on 429 error for POST request

philnash opened the pull request:
Got by default does not retry on POST requests. This changes the retry behaviour to retry across all of the methods that the API uses (GET, POST and DELETE) but only for 429 errors.

Excuse all the other changes, I think they came from the linter during commit. The important lines (right at the bottom) are: 

```diff
    options.retry = {
      limit: this.config.retryLimit || RETRY_LIMIT,
+     methods: ["GET", "POST", "DELETE"],
+     statusCodes: [429]
    };
```

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

philnash committed:
fix(serverless-api): retries on 429 error for POST request

Got by default does not retry on POST requests. This changes the retry behaviour to retry across all of the methods that the API uses (GET, POST and DELETE) but only for 429 errors.

chore(prettier): re-adds prettierrc to the top level

dkundel reviewed (changes requested):
I think the prettier config might have gone lost during the move. Can you re add it to the top level (https://github.com/twilio-labs/serverless-toolkit/blob/master/.prettierrc) and fix the formatting? I'll merge this in then :)

philnash commented:
Added prettierrc, squashed and force pushed up. The diff is better now. Did you have opinions about brackets around single arguments to arrow functions though? 😄
